weave-js/src/components/Panel2/panelTree.ts

```diff
@@ -428,13 +440,23 @@ export const mapPanelsAsync = async (
   if (isGroupNode(fullNode)) {
     const items: {[key: string]: ChildPanelFullConfig} = {};
     for (const key of Object.keys(fullNode.config.items)) {
-      items[key] = await mapPanelsAsync(
-        fullNode.config.items[key],
-        stack,
-        fn,
-        dirtyAction,
-        [...path, key]
-      );
+      if (dirtyAction?.type === 'VarRename' && _.isEqual([...path, key], dirtyAction?.path)) {
```

Comment:
i modified mapPanels because we need to modify the key to update the varName and you can do that from the function passed into mapPanels. If I can get the other way to work we wont need to do this